### DISCUSSION TITLE
ReadOnly mode & hidden features

### DISCUSSION THREAD

**Discussion by giom-l** [kafbat/kafka-ui#343] [Q&A] [answered]
Hello, 

I'd rather opening a discussion here instead of a bug since I don't know if it has been designed as is on purpose or not.

I come from the kafka-ui (I guess as a lot of people) and I'm obviously comparing those 2 :)
I noticed that kafbat-ui in readOnly mode still let appear those 2 buttons on topic details panel (at upper right corner). 
![image](https://github.com/kafbat/kafka-ui/assets/29728596/cb7f0a10-819a-445d-8155-8b6fc478ef23)

![image](https://github.com/kafbat/kafka-ui/assets/29728596/52fd989a-0b04-4922-ae2f-2634283e7b55)

Sure those are disabled and it's OK regarding the readOnly mode feature. 
But shouldn't those be completely masked from the users ?

**Comment by Haarolean** [comment, marked as answer]
Hi, that's absolutely okay to compare the older versions of the app.

From the UX perspective, that's a better approach to show the user that there *is* an action (but not an available one) rather than not showing anything at all. While this might be not that revelant to read-only mode, it's rather preferrable with RBAC.

So, it's quite intended, as managing both the states for r/o and RBAC modes for the every button in the app is quite cumbersome.

Let me know what you think.

**Reply by siegenthalerroger** [reply]
I definitely prefer seeing the available features even if they are deactivated (for RBAC or readOnly mode).
